File: Jay/lv1/134240. 푸드 파이트 대회 /푸드 파이트 대회.kt
class Solution {
    fun solution(food: IntArray): String {
        val leftArray: MutableList<String> = mutableListOf()
        val rightArray: MutableList<String>
    
        for (i in 1 until food.size) {
            var amount = food[i]
            if (amount % 2 != 0) {
                amount -= 1
            }
        
            if (amount <= 1) {
                continue
            }
        
            val foodType = i.toString()
            for (j in 0 until (amount / 2)) {
                leftArray.add(foodType)
            }
        }
    
        rightArray = leftArray.reversed().toMutableList()
        val fullArray = leftArray.toMutableList()
        fullArray.add("0")
        fullArray.addAll(rightArray)
    
        return fullArray.joinToString(separator = "")
    }
}

File: Jay/lv1/134240. 푸드 파이트 대회 /푸드 파이트 대회.swift
import Foundation

func solution(_ food:[Int]) -> String {
    var leftArray: [String] = []
    var rightArray: [String] = []
    
    for i in 1..<food.count {
        var amount = food[i]
        if amount % 2 != 0 {
            amount -= 1
        }
        
        if amount <= 1 {
            continue
        }
        
        let foodType = String(i)
        for _ in 0..<(amount / 2) {
            leftArray.append(foodType)
        }
    }
    
    rightArray = leftArray.reversed()
    var fullArray = leftArray
    fullArray.append("0")
    fullArray += rightArray
    
    let result = fullArray.joined()
    return result
}
